src/store-api/src/region_request.rs

```diff
@@ -501,6 +515,60 @@ impl TryFrom<v1::AddColumnLocation> for AddColumnLocation {
     }
 }
 
+/// Modify a column.
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub struct ModifyColumn {
+    /// Schema of the column to modify.
+    pub column_name: String,
+    /// Column will be modified to this type.
+    pub target_type: ConcreteDataType,
+}
+
+impl ModifyColumn {
+    /// Returns an error if the column to modify is invalid.
+    pub fn validate(&self, metadata: &RegionMetadata) -> Result<()> {
+        let column_meta =
+            metadata
+                .column_by_name(&self.column_name)
+                .context(InvalidRegionRequestSnafu {
+                    region_id: metadata.region_id,
+                    err: format!("column {} not found", self.column_name),
+                })?;
```

Comment:
Avoid constructing the struct as it uses `format`.
```suggestion
                .with_context(|| InvalidRegionRequestSnafu {
                    region_id: metadata.region_id,
                    err: format!("column {} not found", self.column_name),
                })?;
```